### PR TITLE
Introduce optional User.primaryUsername

### PR DESCRIPTION
**Pull request by leogrim**
@andrewconner @stkem @Colin-Lane @eishay 
Funny story, haven't got any merge conflict :sparkles:

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/model/User.scala [hunk 1]
```diff
@@ -59,16 +62,53 @@ object User {
     (__ \ 'userPictureId).formatNullable[Id[UserPicture]] and
     (__ \ 'seq).format(SequenceNumber.format[User]) and
     (__ \ 'primaryEmail).formatNullable[EmailAddress] and
-    (__ \ 'username).format[Username] and
-    (__ \ 'normalizedUsername).format[String]
-  )(User.apply, unlift(User.unapply))
+    (__ \ 'username).formatNullable[Username] and
+    (__ \ 'normalizedUsername).formatNullable[Username]
+  )(User.applyFromDbRow, unlift(User.unapplyToDbRow))
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/model/User.scala:67]: User serialization to json is unchanged.

File: kifi-backend/modules/common/app/com/keepit/model/User.scala [hunk 2]
```diff
@@ -59,16 +62,53 @@ object User {
...
+      user.firstName,
+      user.lastName,
+      user.state,
+      user.pictureName,
+      user.userPictureId,
+      user.seq,
+      user.primaryEmail,
+      user.primaryUsername.map(_.original),
+      user.primaryUsername.map(_.normalized)))
+  }
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/model/User.scala:104]: The User table isn't changed either (well, except for allowing `username` and `normalized_username` to be `NULL`, which I haven't applied the evolution for yet)

**Comment by leogrim** [comment]
Bonus: every instantiation of `User` in tests is now relying on `UserFactory`.

**Comment by andrewconner** [comment]
Wow! Nice

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/LibraryFeedController.scala [hunk 3]
```diff
@@ -11,9 +11,6 @@ import play.api.mvc.{ ResponseHeader, Result }
 
 import scala.concurrent.Future
 
-/**
- * Created by colinlane on 5/22/15.
- */
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/LibraryFeedController.scala:16]: @Colin-Lane make sure you remove these as they're added by IntelliJ

Comment (Colin-Lane) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/LibraryFeedController.scala:16]: Yeah, I fixed my templates a few days ago as it was pointed out to me. I guess one of two had gotten through earlier un-noticed

**Pull request merged**